packages/p2p/src/components/composite-calendar/composite-calendar-mobile.jsx

```diff
@@ -0,0 +1,253 @@
+            setSelectedDateRange({ value: CUSTOM_KEY });
+
+            const value = e.target?.value ? toMoment(e.target.value).format('DD MMM YYYY') : '';
+
+            if (key === 'from') {
+                setFrom(value);
+            }
+
+            if (key === 'to') {
+                setTo(value);
+            }
+        };
+
+        const getMobileFooter = () => {
+            return (
+                <div className='composite-calendar-modal__actions'>
+                    <Button
+                        className='composite-calendar-modal__actions__cancel'
+                        text={localize('Cancel')}
+                        onClick={() => setIsOpen(false)}
+                        has_effect
+                        secondary
+                        large
+                    />
+                    <Button
+                        className='composite-calendar-modal__actions__ok'
+                        text={localize('OK')}
+                        onClick={applyDateRange}
+                        has_effect
+                        primary
+                        large
+                    />
+                </div>
+            );
+        };
+
+        const onDateRangeChange = _date_range => {
+            setSelectedDateRange(
+                duration_list.find(range => _date_range && range.value === _date_range.value) || _date_range
+            );
+        };
+
+        const openDialog = () => {
+            setSelectedDateRange(applied_date_range);
+            setIsOpen(true);
+        };
+
+        const today = toMoment().format('YYYY-MM-DD');
+        const max_date = to ? toMoment(to, 'DD MMM YYYY').format('YYYY-MM-DD') : today;
+        const min_date = from && toMoment(from, 'DD MMM YYYY').format('YYYY-MM-DD');
+
+        return (
+            <React.Fragment>
+                <div className='composite-calendar__input-fields composite-calendar__input-fields--fill'>
+                    <InputField
+                        data_testid='dt_calendar_input'
+                        current_focus={current_focus}
+                        is_read_only={true}
+                        icon={() => <Icon icon='IcCalendarDatefrom' className='inline-icon' />}
+                        onClick={openDialog}
+                        setCurrentFocus={setCurrentFocus}
+                        value={applied_date_range.label}
+                    />
+                </div>
+                <MobileDialog
+                    portal_element_id='modal_root'
+                    title={localize('Please select duration')}
+                    visible={is_open}
+                    has_content_scroll
+                    onClose={() => setIsOpen(false)}
+                    content_height_offset='94px'
+                    footer={getMobileFooter()}
+                >
+                    <div className='composite-calendar-modal'>
+                        <div className='composite-calendar-modal__radio-group'>
```

Comment:
If we used the radio group component that we have, could you somehow apply it here? or is it too much of a change

Reply:
i did try using the radio group here, but it makes a lot of change and radio group currently accepts only arrays and that doesn't work with the custom value. so again for the custom it has to be changed to an array.